docs/rust_analyzer.md

```diff
@@ -51,7 +51,7 @@ A list of `rust_analyzer` compatible targets can be found by usign the following
 bazel query 'kind("rust_*library|rust_binary", //...:all)'
 ```
 
-Note that visibility rules apply.
+Note: __All `rust_*` targets provided to the root rust_analyzer must have `//visibility:public`.__
```

Comment:
It's actually not true - all rust targets provided to the rust_analyzer target have to be visible to the rust_analyzer target. That's a slightly weaker restriction.

Reply:
When I exposed the targets to the analyzer rule, either individually or via a package_group rule, Bazel failed analysis because of a circular dependency. If you know of another way to have two targets reference each other explicitly, please update the docs.